core-common/src/main/java/org/zalando/nakadi/domain/StrictJsonParser.java

```diff
@@ -139,6 +139,9 @@ private static Object readNumberTillTheEnd(final char value, final StringTokeniz
             throw syntaxError("Unexpected symbol '" + value + "'", tokenizer);
         }
         final int start = tokenizer.getCurrentPosition() - 1;
+        if (!Character.toString(tokenizer.value.charAt(start)).matches("\\d|-")) {
```

Comment:
Performance-wise regex looks like too heavy here.  We could just check with `>= '0' && <= '9'` or something like that.
Alternatively we might not even go into this function if we see it start with the dot.

Reply:
Would you still think that Regex that has a very small syntax tree and is applied against a single character value could be overwhelming for this part?

Reply:
For one the regexp will be compiled every time, or am I missing something?

Reply:
Yes, in this case, it does compile every time but there seem to be workarounds.